File: rustbox/tests/languages/lang_cpp/test_5_mle.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    size_t size = 1;
    while (1) {
        try {
            vector<int> arr(size);
            size *= 2;
        } catch (bad_alloc& e) {
            cout << "Memory allocation failed at size = " << size << endl;
            break;
        }
    }
    return 0;
}

File: rustbox/src/tests/languages/test_files/lang_java/test_5_mle.java
import java.util.ArrayList;
import java.util.List;

public class test_5_mle {
    public static void main(String[] args) {
        List<List<Integer>> data = new ArrayList<>();
        int size = 1;
        
        try {
            while (true) {
                List<Integer> arr = new ArrayList<>();
                for (int i = 0; i < size; i++) {
                    arr.add(0);
                }
                data.add(arr);
                size *= 2;
                if (size <= 0) break;
            }
        } catch (OutOfMemoryError e) {
            System.out.println("Memory allocation failed");
        } catch (Exception e) {
            System.out.println("Error: " + e.getMessage());
        }
    }
}